orchestrator/src/main/kotlin/com/ntgjvmagent/orchestrator/service/ToolService.kt

```diff
@@ -58,4 +83,131 @@ class ToolService(
         tool.markDeleted()
         repo.save(tool)
     }
+
+    fun insertTool(
+        toolCallbacks: List<ToolCallback>,
+        request: ToolRequestDto,
+    ) {
+        if (toolCallbacks.isEmpty()) {
+            return
+        }
+
+        val existingToolNames = repo.findAllByActiveTrue().map { it.name }
+        for (toolCallback in toolCallbacks) {
+            val toolDefinition = toolCallback.toolDefinition
+            val toolName = toolDefinition.name()
+            if (existingToolNames.contains(toolName)) {
```

Comment:
We need to check against inactive ones as well (all undeleted ones) in the same source. 
We may need to allow the same name in different sources.